cmake/scripts/buildinfo.cmake

```
@@ -19,19 +19,32 @@ if (NOT ETH_BUILD_PLATFORM)
 	set(ETH_BUILD_PLATFORM "unknown")
 endif()
 
-# Logic here: If prerelease.txt exists but is empty, it is a non-pre release.
-# If it does not exist, create our own prerelease string
+
 if (EXISTS ${ETH_SOURCE_DIR}/prerelease.txt)
-	file(READ ${ETH_SOURCE_DIR}/prerelease.txt SOL_VERSION_PRERELEASE)
-	string(STRIP "${SOL_VERSION_PRERELEASE}" SOL_VERSION_PRERELEASE)
+	message(FATAL_ERROR "The usage of the file prerelease.txt is deprecated. Please use the file cmake/prerelease.lock instead.")
+endif()
+
+if (SOLC_FORCE_RELEASE)
+	set(SOL_VERSION_PRERELEASE, "")
 else()
```

You can use `elseif` to make the `if` less nested:
```suggestion
elseif(SOL_PRERELEASE_STRING)
```
**EDIT**: Replaced `SOLC_` prefix with `SOL_` (see comments below).